catalog-connectors/airflow/airflow-package-catalog-connector/airflow_package_catalog_connector/airflow_package_catalog_connector.py

```diff
@@ -0,0 +1,245 @@
+        airflow_package_download_url = catalog_metadata['airflow_package_download_url']
+        # extract the package name, e.g. 'apache_airflow-1.10.15-py2.py3-none-any.whl'
+        airflow_package_name = Path(urlparse(airflow_package_download_url).path).name
+
+        # tmp_archive_dir is used to store the downloaded archive and as working directory
+        if hasattr(self, 'tmp_archive_dir'):
+            # if the directory exists remove it in case the archive content has changed
+            shutil.rmtree(self.tmp_archive_dir.name, ignore_errors=True)
+        self.tmp_archive_dir = Path(mkdtemp())
+
+        try:
+            self.log.debug(f'Downloading Apache Airflow package from \'{airflow_package_download_url}\' ...')
+
+            # download archive; abort after 30 seconds
+            response = requests.get(airflow_package_download_url,
+                                    timeout=30,
+                                    allow_redirects=True)
+            if response.status_code != 200:
+                # download failed. Log error and abort processing
+                self.log.error(f'Download of archive \'{airflow_package_download_url}\' '
+                               'failed. HTTP response code: {response.status_code}')
+                return operator_key_list
+
+            # save downloaded archive
+            archive = str(self.tmp_archive_dir / airflow_package_name)
+            self.log.debug(f'Saving downloaded archive in \'{archive}\' ...')
+            with open(archive, 'wb') as archive_fh:
+                archive_fh.write(response.content)
+
+            # extract archive
+            self.log.debug(f'Extracting Airflow archive \'{archive}\' ...')
+            with zipfile.ZipFile(archive, 'r') as zip_ref:
+                zip_ref.extractall(self.tmp_archive_dir)
+
+            # delete archive
+            self.log.debug(f'Deleting downloaded Airflow archive \'{archive}\' ...')
+            os.remove(archive)
+
+            # Locate Python scripts that are stored in the 'airflow/operators' directory
+            python_scripts = [str(s) for s in self.tmp_archive_dir.glob('airflow/operators/*.py')]
+
+            #
+            # Identify Python scripts that define classes that extend the
+            # airflow.models.BaseOperator class
+            #
+            scripts_with_operator_class = []  # list of str, containing Python script names with operator definitions
+            extends_baseoperator = []  # list of str, containing classes that extend BaseOperator
+            classes_to_analyze = {}
+            imported_operator_classes = []  # list of str, identifying imported operator classes
+            offset = len(str(self.tmp_archive_dir)) + 1
+            script_count = 0  # used for stats collection
+            class_count = 0   # used for stats collection
+            operator_class_count = 0  # used for stats collection
```

Comment:
I think you could just use `len(extends_baseoperator)` instead of having this additional variable